src/main/scala/org/bitcoins/core/protocol/script/ScriptSignature.scala

```diff
@@ -39,15 +39,21 @@ sealed trait ScriptSignature extends NetworkElement with BitcoinSLogger {
 
 
   /**
-    * Derives the hash type for a given digitalSignature
+    * Derives the hash types for every digitalSignature in the ScriptSignature object
     *
-    * @param digitalSignature
     * @return
     */
-  def hashType(digitalSignature: ECDigitalSignature) : HashType = {
-    digitalSignature match {
-      case EmptyDigitalSignature => SIGHASH_ALL(Int32.one)
-      case sig : ECDigitalSignature => HashType(Seq(digitalSignature.bytes.last))
+  def hashTypes : Seq[HashType] = {
```

Comment:
Ok let's remove it then, no point in keeping this for the added complexity. Although I think it would be handy to have a function some where to get the hash type of a digital signature, perhaps `DERSignatureUtil`? I guess we have to be careful of bitcoin digital signatures vs standard digital signatures used in crypto libraries